packages/controller/src/index.ts

```diff
@@ -17,39 +13,52 @@ import {
   ResponseCodes,
   ConnectReply,
   ProbeReply,
-  Modal,
   ControllerOptions,
-  ControllerThemePresets,
-  ColorMode,
   PaymasterOptions,
-  // Prefund,
   ConnectError,
+  Profile,
+  IFrames,
 } from "./types";
-import { createModal } from "./modal";
-import { defaultPresets } from "./presets";
-import { NotReadyToConnect } from "./errors";
-import { KEYCHAIN_URL, RPC_SEPOLIA } from "./constants";
+import { KeychainIFrame, ProfileIFrame } from "./iframe";
+import { NotReadyToConnect, ProfileNotReady } from "./errors";
+import { RPC_SEPOLIA } from "./constants";
 
 class Controller {
-  private url: URL;
   private policies: Policy[];
   private paymaster?: PaymasterOptions;
-  private connection?: Connection<Keychain>;
-  private modal?: Modal;
   public keychain?: AsyncMethodReturns<Keychain>;
+  public profile?: AsyncMethodReturns<Profile>;
+  private iframes: IFrames;
   public rpc: URL;
   public account?: AccountInterface;
 
   constructor({
     policies,
     url,
+    profileUrl,
```

Comment:
Mmm do we need to make this user configurable? Could it just be const?

Reply:
Good point. Yes, I agree that there's no need for game dev to set this url. But we want it as a Controller dev to switch between localhost and production API I think. Maybe we can introduce env to Controller but I'm not keen to do so since it's a library not app.